source/plugin/items/creating.rst

```diff
@@ -107,14 +107,9 @@ An example is shown below:
     
     public void spawnItem(ItemStack superMegaAwesomeSword, Location<World> spawnLocation) {
         Extent extent = spawnLocation.getExtent();
-        Optional<Entity> optional = extent
-            .createEntity(EntityTypes.ITEM, spawnLocation.getPosition());
-        if (optional.isPresent()) {
-            Entity item = optional.get();
-            item.offer(Keys.REPRESENTED_ITEM, superMegaAwesomeSword.createSnapshot());
-            extent.spawnEntity(item, Cause.source(EntitySpawnCause.builder()
-                .entity(item).type(SpawnTypes.PLUGIN).build()).build());
-        }
+        Entity item = extent.createEntity(EntityTypes.ITEM, spawnLocation.getPosition());
+        item.offer(Keys.REPRESENTED_ITEM, superMegaAwesomeSword.createSnapshot());
+        extent.spawnEntity(item);
```

Comment:
Likewise here, need to add the `SpawnType` to the context.

Reply:
Can wrap with:
```java
try (CauseStackManager.StackFrame frame = Sponge.getCauseStackManager().pushStackFrame()) {
    frame.addContext(EventContextKeys.SPAWN_TYPE, SpawnTypes.PLACEMENT);
    extent.spawnEntity(item);
}
```

Reply:
Fixed